modules/unsupported/geojson-core/src/main/java/org/geotools/data/geojson/GeoJSONReader.java

```diff
@@ -436,6 +451,24 @@ private SimpleFeature getNextFeature(ObjectNode node) throws IOException {
         return feature;
     }
 
+    private String getId(ObjectNode node) {
+        String newId = null;
+        if (node.has(ID)) {
+            if (idPrefix != null) {
+                newId = idPrefix + "." + node.get(ID).asText();
+            } else {
+                newId = node.get(ID).asText();
```

Comment:
What happens if the `ID` attributes are not unique?

Reply:
My first reaction would have been "then the GeoJSON is broken". But it turns out the specification even allows the id to be missing and does not talk about uniqueness.

However this can be restricted by specifications that build on top of GeoJSON. For example, the [STAC-item specification](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#item-fields) mandates "The ID should be unique within the [Collection](https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md) that contains the Item."

So it seems we likely need some id generation strategies that a caller can use:

- Existing one, file name + progressive, for random GeoJSON files that use the full liberty of the spec.
- Using exactly the id found in the features, for contexts where uniqueness is guaranteed
- Also keep the file name + id, to make the ids "more globally unique" (needed for the STAC datastore, to make it behave like other GeoServer stores).

Reply:
Implemented, as described.